vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssNodeSpecModel.kt

```diff
@@ -42,25 +42,28 @@ import org.eclipse.kuksa.vsscore.model.name
 import org.eclipse.kuksa.vsscore.model.parentClassName
 import org.eclipse.kuksa.vsscore.model.parentKey
 import org.eclipse.kuksa.vsscore.model.variableName
+import org.eclipse.kuksa.vssprocessor.spec.VssNodeProperty.*
 import kotlin.reflect.KClass
 import kotlin.reflect.KProperty1
 import kotlin.reflect.full.declaredMemberProperties
 
-// Reflects the VSS file as a model and is filled via reflection. That is why the variable names
-// should exactly match the names inside the VSS file and be of a string type.
 internal class VssNodeSpecModel(
-    override var uuid: String = "",
     override var vssPath: String = "",
-    override var description: String = "",
-    override var type: String = "",
-    override var comment: String = "",
-    @Suppress("MemberVisibilityCanBePrivate") var datatype: String = "",
+    val vssNodeProperties: Set<VssNodeProperty>,
 ) : VssNode, SpecModel<VssNodeSpecModel> {
+
+    private val propertyNameToNodePropertyMap = vssNodeProperties.associateBy { it.nodePropertyName }
+
+    override var uuid: String = propertyNameToNodePropertyMap["uuid"]?.nodePropertyValue ?: ""
+    override var type: String = propertyNameToNodePropertyMap["type"]?.nodePropertyValue ?: ""
+    override var description: String = propertyNameToNodePropertyMap["description"]?.nodePropertyValue ?: ""
+    override var comment: String = propertyNameToNodePropertyMap["comment"]?.nodePropertyValue ?: ""
+    var datatype: String = propertyNameToNodePropertyMap["datatype"]?.nodePropertyValue ?: ""
```

Comment:
kept for compatibility with old code

Reply:
OK for now. Still a good step in the right direction :)